common/persistence/sql/execution.go

```diff
@@ -601,13 +601,81 @@ func (m *sqlExecutionStore) DeleteWorkflowExecution(
 ) error {
 	namespaceID := primitives.MustParseUUID(request.NamespaceID)
 	runID := primitives.MustParseUUID(request.RunID)
-	_, err := m.Db.DeleteFromExecutions(ctx, sqlplugin.ExecutionsFilter{
-		ShardID:     request.ShardID,
-		NamespaceID: namespaceID,
-		WorkflowID:  request.WorkflowID,
-		RunID:       runID,
+	return m.txExecute(ctx, "DeleteWorkflowExecution", func(tx sqlplugin.Tx) error {
+		_, err := tx.DeleteAllFromChildExecutionInfoMaps(ctx, sqlplugin.ChildExecutionInfoMapsAllFilter{
+			ShardID:     request.ShardID,
+			NamespaceID: namespaceID,
+			WorkflowID:  request.WorkflowID,
+			RunID:       runID,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to execute DeleteAllFromChildExecutionInfoMaps: %w", err)
```

Comment:
Hi @prathyushpv, I would move this formatting into `tx.DeleteAll function` and at this level only adding that it was called from DeleteWorkflowExecution. This way log message will not only contain what function failed but also from where it was called from. But I am new to Go, may be Go does it automatically. @alexshtin what do you think?

Reply:
That makes sense, but these DeleteAll functions are just one line functions executing an sql qeury and returning result. Also, we are wrapping the error here, so we are not losing the original error.